crates/reth/node/src/gossip/package.rs

```diff
@@ -0,0 +1,359 @@
+//! Gossip package and message types.
+//!
+//! Right now, we only support [`AlpenGossipPackage`] and [`AlpenGossipMessage`].
+
+use std::mem;
+
+use alloy_primitives::{
+    bytes::{Buf, BufMut, BytesMut},
+    eip191_hash_message,
+};
+use alloy_rlp::{Decodable, Encodable};
+use eyre::{ensure, eyre, Result};
+use reth_primitives::Header;
+use strata_primitives::{
+    buf::Buf64,
+    crypto::{sign_schnorr_sig, verify_schnorr_sig},
+    Buf32,
+};
+
+/// Size of the sequence number in bytes.
+const SEQ_NO_SIZE: usize = mem::size_of::<u64>();
+
+/// Gossip message types.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct AlpenGossipMessage {
+    /// Block [`Header`].
+    header: Header,
+
+    /// Sequence number.
+    seq_no: u64,
```

Comment:
Is this `seq_no` used to prevent receiving of duplicate message from a peer?

I don't see it being used though(except for encoding, decoding).